src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -950,6 +968,7 @@ private void AbortStream(int streamId, IOException error)
         {
             if (_streams.TryGetValue(streamId, out var stream))
             {
+                stream.DecrementActiveStreamCount();
```

Comment:
Are you sure about this one? It looks like it's only called from here:
https://github.com/aspnet/AspNetCore/blob/ef99d81d681c6d3ddf8af9c7c49dde7f4afb5170/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs#L221
I'd think we'd not have incremented yet if Http2StreamErrorException was thrown on that code path.

Reply:
I'll debug and see if it's hit. If that's the case, I'll add a debug assert.

Reply:
It looks like the stream count was already incremented, so it is necessary. We increment after we decode the headers and call StartStream, which occurs before we throw an Http2StreamErrorException.